src/PhpSpec/Matcher/ThrowMatcher.php

```diff
@@ -91,114 +90,134 @@ public function negativeMatch($name, $subject, array $arguments)
     }
 
     /**
-     * @param callable $callable
-     * @param array    $arguments
-     * @param null     $exception
+     * @param callable           $callable
+     * @param array              $arguments
+     * @param null|object|string $exception
      *
      * @throws \PhpSpec\Exception\Example\FailureException
      * @throws \PhpSpec\Exception\Example\NotEqualException
      */
     public function verifyPositive($callable, array $arguments, $exception = null)
     {
+        $exceptionThrown = null;
+
         try {
             call_user_func_array($callable, $arguments);
         } catch (\Exception $e) {
-            if (null === $exception) {
-                return;
-            }
+            $exceptionThrown = $e;
+        } catch (\Throwable $e) {
+            $exceptionThrown = $e;
+        }
+
+        if (!$exceptionThrown) {
+            throw new FailureException('Expected to get exception, none got.');
```

Comment:
Maybe this message should reflect that it might not be an exception? `Expected to get exception / throwable...`?